exonum/src/helpers/config.rs

```diff
@@ -66,3 +68,64 @@ fn do_save<T: Serialize>(value: &T, path: &Path) -> Result<(), Error> {
     file.write_all(value_toml.to_string().as_bytes())?;
     Ok(())
 }
+
+/// Structure that handles work with config file in runtime.
+#[derive(Debug)]
+pub struct ConfigManager {
+    handle: thread::JoinHandle<()>,
+    tx: mpsc::Sender<ConfigRequest>,
+}
+
+/// Messages for ConfigReader.
+#[derive(Debug)]
+pub enum ConfigRequest {
+    /// Request for connect list update in config file.
+    UpdateConnectList(ConnectListConfig),
+}
+
+impl ConfigManager {
+    /// Creates a new `ConfigManager` instance for the given path.
+    pub fn new(path: String) -> Self {
+        let (tx, rx) = mpsc::channel();
+        let handle = thread::spawn(move || {
+            info!("ConfigManager started");
+            for request in rx {
+                match request {
+                    ConfigRequest::UpdateConnectList(connect_list) => {
+                        info!("Updating connect list. New value: {:?}", connect_list);
+
+                        let res = Self::update_connect_list(connect_list, &path);
+
+                        if let Err(ref error) = res {
+                            error!("Unable to update config: {}", error);
+                        }
+                    }
+                }
+            }
+            info!("ConfigManager stopped");
+        });
+
+        ConfigManager { handle, tx }
+    }
+
+    /// Stores updated connect list in file system.
+    pub fn store_connect_list(&self, connect_list: ConnectListConfig) {
+        self.tx
+            .send(ConfigRequest::UpdateConnectList(connect_list))
+            .expect("Can't write");
```

Comment:
I know the possibility of error here, but let's use more descriptive message like `Can't  send message to ConfigManager thread`